safe-global/safe-wallet-web
# fix: Watch nested safe transactions

## What it solves

Resolves #4025 

## How this PR fixes it

- Checks for each pending transactions if it is a nested safe transaction and in case it is it polls the CGW until it returns a `txHash` which then starts the usual tx watcher
- Removes `waitForTx` calls from dispatchers and instead runs the `useTxMonitor` hook whenever a pending tx is added

## How to test it

1. Open a 1/n safe that has another safe as an owner
2. Connect to the parent safe
3. Create and execute a transaction
4. Observe in the networks tab that the CGW is called every 15s
5. Once the parent safe executes the transaction the pending state should resolve

## Screenshots

## Checklist
